packages/mdc-menu/simple/foundation.js

```diff
@@ -399,69 +335,202 @@ class MDCSimpleMenuFoundation extends MDCFoundation {
-      if (extendsBeyondRightBounds && leftOverflow < rightOverflow) {
-        horizontal = 'left';
+  /**
+   * @param {Corner} corner Origin corner of the menu.
+   * @return {number} Horizontal offset of menu origin corner from corresponding anchor corner.
+   * @private
+   */
+  getHorizontalOriginOffset_(corner) {
+    const {anchorRect} = this.measures_;
+    const horizontalAlignment = (corner & CornerBit.RIGHT) ? 'right' : 'left';
+    let x = 0;
+    if (horizontalAlignment === 'right') {
+      const rightOffset = (this.anchorCorner_ & CornerBit.RIGHT) ?
+        anchorRect.width - this.anchorMargin_.left : this.anchorMargin_.right;
+      x = rightOffset;
+    } else {
+      const leftOffset = (this.anchorCorner_ & CornerBit.RIGHT) ?
+        anchorRect.width - this.anchorMargin_.right : this.anchorMargin_.left;
+      x = leftOffset;
+    }
+    return x;
+  }
+
+  /**
+   * @param {Corner} corner Origin corner of the menu.
+   * @return {number} Vertical offset of menu origin corner from corresponding anchor corner.
+   * @private
+   */
+  getVerticalOriginOffset_(corner) {
+    const {anchorRect, viewport, viewportMargin, anchorHeight, menuHeight} = this.measures_;
+    const verticalAlignment = (corner & CornerBit.BOTTOM) ? 'bottom' : 'top';
+    const {MARGIN_TO_EDGE} = MDCSimpleMenuFoundation.numbers;
+    const avoidVerticalOverlap = this.anchorCorner_ & CornerBit.BOTTOM;
+    const canOverlapVertically = !avoidVerticalOverlap;
+    let y = 0;
+
+    if (verticalAlignment === 'bottom') {
+      y = avoidVerticalOverlap ? anchorRect.height - this.anchorMargin_.top : -this.anchorMargin_.bottom;
+      // adjust for when menu can overlap anchor, but too tall to be aligned to bottom
+      // anchor corner. Bottom margin is ignored in such cases.
+      if (canOverlapVertically && menuHeight > viewportMargin.top + anchorHeight) {
+        y = -(Math.min(menuHeight, viewport.height - MARGIN_TO_EDGE) - (viewportMargin.top + anchorHeight));
+      }
+    } else {
+      y = avoidVerticalOverlap ? (anchorRect.height + this.anchorMargin_.bottom) : this.anchorMargin_.top;
+      // adjust for when menu can overlap anchor, but too tall to be aligned to top
+      // anchor corners. Top margin is ignored in that case.
+      if (canOverlapVertically && menuHeight > viewportMargin.bottom + anchorHeight) {
+        y = -(Math.min(menuHeight, viewport.height - MARGIN_TO_EDGE) - (viewportMargin.bottom + anchorHeight));
       }
-    } else if (extendsBeyondLeftBounds && rightOverflow < leftOverflow) {
-      horizontal = 'right';
     }
+    return y;
+  }
 
+  /**
+   * @param {Corner} corner Origin corner of the menu.
+   * @return {number} Maximum height of the menu, based on available space. 0 indicates should not be set.
+   * @private
+   */
+  getMenuMaxHeight_(corner) {
+    let maxHeight = 0;
+    const {viewportMargin} = this.measures_;
+    const verticalAlignment = (corner & CornerBit.BOTTOM) ? 'bottom' : 'top';
```

Comment:
I'm not sure there's really a point to this variable here, could just inline `corner & CornerBit.BOTTOM` below in the `if`?

Also, I'm not sure I understand why we're only computing max height if bottom-aligned? Shouldn't this be equally relevant to top aligned? Or is that what the maybe-a-TODO below is about?

Reply:
I think that having a variable there makes it little bit more readable.
top aligned means you can overlap the anchor, in that case max-height of the menu is driven by css from vh - 32 style.

Reply:
Add a comment above the outer if below to indicate that the alternative case is handled by CSS (if I'm understanding correctly)